### ISSUE TITLE
[HOLD #53760][$250] Create expense - No animation when submitting expenses as a new user

### ISSUE THREAD

**Issue by vincdargento** [Expensify/App#54470]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.77-6
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** N/A
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition: You should login as a new user.

1. Open https://staging.new.expensify.com/
2. Click on the green "+" button.
3. Submit a manual expense

## Expected Result:
A fun illustration (with short and clear directions) should be displayed to the empty state “Submits to” screen if the user does not have any recent contacts (as in PR #42413). If there are any recents in the list, the illustration shouldn't be displayed.

## Actual Result:
No animation when submitting expenses as a new user, just empty page.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/c7f22e83-3e01-4da9-980e-4fdcf2700a0a

![bug](https://github.com/user-attachments/assets/d06a4ccb-db30-4882-aa2d-838821a8554b)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021871676942933585608</li>
        <li>Upwork Job ID: 1871676942933585608</li>
        <li>Last Price Increase: 2024-12-24</li>
<li>Automatic offers: </li>
<ul>
<li>huult | Contributor | 105506456</li>
</ul></li>
    </ul>
</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Create expense - No animation when submitting expenses as a new user

### What is the root cause of that problem?
We are only handling split, pay, and submit
https://github.com/Expensify/App/blob/126d92ab5cbb401efc0a5527faba244d04d48971/src/components/EmptySelectionListContent.tsx#L21
https://github.com/Expensify/App/blob/126d92ab5cbb401efc0a5527faba244d04d48971/src/components/EmptySelectionListContent.tsx#L32-L34
so blank is displayed for iouType create (create expense from global FAB) and also invoice (from send invoice)
### What changes do you think we should make in order to solve the problem?

We need to add cases of `INVOICE` and `CREATE`:
1. add both types [here](https://github.com/Expensify/App/blob/126d92ab5cbb401efc0a5527faba244d04d48971/src/components/EmptySelectionListContent.tsx#L21)
2. Add the respective copies [here](https://github.com/FitseTLT/App/blob/126d92ab5cbb401efc0a5527faba244d04d48971/src/languages/en.ts#L588-L592)

If style changes are needed we can apply discussing with the design team
Note: We are still using submit in submit it to someone whisper action from tracked expense so we can't remove submit.
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
We can test EmptySelectionListContent displays correctly for iouTypes `INVOICE` and `CREATE`

### What alternative solutions did you explore? (Optional)

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

No animation when submitting expenses as a new user

### What is the root cause of that problem?

In this case, the `iouType` is `create` and is not included in the list of [`CONTENT_TYPES`](https://github.com/Expensify/App/blob/3f0b2f8bdff787b6e1af0f326b47084f1f426e7e/src/components/EmptySelectionListContent.tsx#L21), so the empty page will not be displayed

https://github.com/Expensify/App/blob/ba6168a214b03d23d4845374b3c338cb86917e95/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L449

https://github.com/Expensify/App/blob/3f0b2f8bdff787b6e1af0f326b47084f1f426e7e/src/components/EmptySelectionListContent.tsx#L32-L34

### What changes do you think we should make in order to solve the problem?


We have many similar cases, so I think we should create a common empty component to ensure consistency.

https://github.com/Expensify/App/blob/3f0b2f8bdff787b6e1af0f326b47084f1f426e7e/src/components/EmptySelectionListContent.tsx#L28

We created a new prop to determine whether to use the default empty state UI and define a common empty UI, and made the `contentType` prop optional

```diff
  function EmptySelectionListContent({contentType, isDefault}: EmptySelectionListContentProps) {
      const styles = useThemeStyles();
      const {translate} = useLocalize();

      if (!isContentType(contentType)) {
          if (isDefault) {
              return (
                  <View style={[styles.flex1, styles.overflowHidden, styles.minHeight65]}>
                      <BlockingView
                          icon={Illustrations.TurtleInShell}
                          iconWidth={variables.emptyListIconWidth}
                          iconHeight={variables.emptyListIconHeight}
                          title={'Nothing to Display'}
                          subtitle={'There’s currently nothing to display here.'}
                          subtitleStyle={styles.textSupporting}
                          containerStyle={styles.pb10}
                          contentFitImage="contain"
                      />
                  </View>
              );
          }
          return null;
      }
      ...
  }
```

And anywhere we want to use the default empty state, such as this one

https://github.com/Expensify/App/blob/ba6168a214b03d23d4845374b3c338cb86917e95/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx#L449

Update to:

```diff
  listEmptyContent={
        <EmptySelectionListContent
            contentType={iouType}
            isDefault
        />
    }
```

<details>
<summary>POC</summary>

- <img width="1512" alt="Screenshot 2024-12-26 at 15 55 38" src="https://github.com/user-attachments/assets/9cae1b81-fce6-4a73-8993-92762b8fe275" />

</details>

I think the UI should be confirmed by the design team and updated when creating the PR (like: content, image, scrollview....)

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by huult** [comment]
@thesahindia here you are: 

- <img width="525" alt="Screenshot 2024-12-30 at 22 17 23" src="https://github.com/user-attachments/assets/2027f760-2f4a-4fb0-acae-84fa0ebd86d1" />

**Comment by huult** [comment]
@thesahindia Yes, I will provide it later. thank you!

**Comment by puneetlath** [comment]
I also like @huult's more global solution.

**Comment by puneetlath** [comment]
@thesahindia can you take that question?

**Comment by shawnborton** [comment]
We might want to do nothing here and wait until this one hits production: https://github.com/Expensify/App/pull/53760

I think the idea is that for all new users, they will at least have their own personal space as a contact to use in this particular screen. Is that right @trjExpensify @Expensify/design ?

**Comment by dannymcclain** [comment]
Ah yeah good call, that sounds right to me. I'm down to wait on that one and reassess once we see it live. We may still want to add something there but maybe something a bit smaller than what we previously had (I remember there being a lot of issues with viewport sizes and keyboards and stuff with that illustration + message).

**Comment by trjExpensify** [comment]
>I think the idea is that for all new users, they will at least have their own personal space as a contact to use in this particular screen. Is that right @trjExpensify @Expensify/design ?

The selfDM doesn't get created in all cases now, since a couple of days ago. But yeah, they'd have at least one of the selfDM, whoever invited them, or the workspace chat in all cases.

**Comment by sakluger** [comment]
> The selfDM doesn't get created in all cases now, since a couple of days ago. But yeah, they'd have at least one of the selfDM, whoever invited them, or the workspace chat in all cases.

So you're saying that we would still need an animation for users with no self DM, thus we should leave this GH issue open?

**Comment by trjExpensify** [comment]
Not really, because everyone should have at least one participant in the list. I could have sworn we're also working on contacts import, which wouldn't that potentially go here?

**Comment by shawnborton** [comment]
Yeah, we had to revert that one but it should be coming back soon.

So I think my vote would be to do nothing for now, and let these new changes hit the product. Then we can re-evaluate the new user experience from there.

**Comment by trjExpensify** [comment]
Sounds good to me!

**Comment by puneetlath** [comment]
Does that mean we should take this off hold?

**Comment by shawnborton** [comment]
Maybe we do nothing here, because I think the idea is to now pre-fill the `To` selector... does that sound right @trjExpensify?

**Comment by trjExpensify** [comment]
Yeah, that PR is pretty close. That said, if you don't have a group workspace set as your activePolicyID, we won't skip the selector. I'm not passionate about an illustration though tbh, it's never going to be completely blank.

**Comment by shawnborton** [comment]
Cool, I say we just close this then.

**Comment by huult** [comment]
@puneetlath We opened a draft [PR](https://github.com/Expensify/App/pull/54735) and based the details on [this SO](https://stackoverflowteams.com/c/expensify/questions/8590/8591#8591:~:text=Dupe%20issues%20and%20issued%20closed).

If a contributor has been hired for a job and we decide to close the job before it is successfully completed:
	•	50% payment to the contributor and C+ if the contributor was 🎀👀🎀 by a C+ or hired/assigned.
	•	100% payment if the contributor drafted a PR and requested a review.
	•	100% payment to C+ if they reviewed the PR.

One caveat here might be if the hired contributor wasn’t working on their PR with urgency.

Can I receive a payment?

**Comment by sakluger** [comment]
@huult thank you for your comment. It looks like you are eligible for full payment based on the fact that you were assigned to the issue, opened the PR, and requested a review.

I sent you an upwork contract, please let me know once you've accepted the offer: https://www.upwork.com/nx/wm/offer/106011255

Please also close the PR since we have decided not to merge it.

**Comment by huult** [comment]
@sakluger I accepted your offer, thanks

**Comment by sakluger** [comment]
Paid! Thanks again everyone.

**Comment by sakluger** [comment]
Whoops, sorry I missed this! And thanks @puneetlath for reopening.

@huult I've sent an Upwork offer for 50% payment: https://www.upwork.com/nx/wm/offer/106176094. Please let me know once you've accepted. Thanks!

**Comment by huult** [comment]
@sakluger 

@thesahindia is asking you https://github.com/Expensify/App/issues/54470#issuecomment-2640867712 . I have already received your payment.

**Comment by sakluger** [comment]
Sorry about that, I got mixed up. Payment summary incoming.

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @huult $250, paid via Upwork
Contributor+: @thesahindia $125, please request on Newdot

**Comment by garrettmknight** [comment]
$125 approved for @thesahindia
